File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    result = [] # 명예의 전당
    answer = [] # 발표 점수

    for i in score:
        result.append(i)
        if len(result) > k:
            result.remove(min(result))
        answer.append(min(result))

    return answer

print(solution(3, [10, 100, 20, 150, 1, 100, 200]))

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.*;

class Solution {
    public int[] solution(int k, int[] score) {
        List<Integer> result = new ArrayList<Integer>();
        int[] answer = new int[score.length];
        
        for(int i=0; i<score.length; i++) {
            result.add(score[i]);
            if(result.size() > k) {
                result.remove(Integer.valueOf(Collections.min(result)));
            }
            answer[i] = Collections.min(result);
        }
        
        return answer;
    }
}